pkg/test/inject_test.go

```diff
@@ -0,0 +1,103 @@
+package test_test
+
+import (
+	"errors"
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+	"zotregistry.io/zot/pkg/test"
+)
+
+var (
+	errKey1    = errors.New("key1 not found")
+	errKey2    = errors.New("key2 not found")
+	errNotZero = errors.New("not zero")
+	errCall1   = errors.New("call1 error")
+	errCall2   = errors.New("call2 error")
+)
+
+func foo() error {
+	fmap := map[string]string{"key1": "val1", "key2": "val2"}
+
+	_, ok := fmap["key1"] // should never fail
+	if !test.CheckOk(ok) {
```

Comment:
As an alternative approach, ideally test would be an interface with different implementations for test cases compared to the production code.

Reply:
Fixed that way now.